pytket/extensions/cutensornet/structured_state/simulation.py

```diff
@@ -73,48 +74,68 @@ def simulate(
         circuit: The pytket circuit to be simulated.
         algorithm: Choose between the values of the ``SimulationAlgorithm`` enum.
         config: The configuration object for simulation.
+        compilation_params: Experimental feature. Defaults to no compilation.
+            Parameters currently not documented.
```

Comment:
This is now hiding two options that I have decided to deprecate:
- `sort_gates` that was always applied when calling `simulate`. When set to True, this is done by `_get_sorted_gates` (see its docstring). This code is not immediately compatible with classical operations and, although it could be adapted, this is not a priority. I am not convinced it was making a noticeable impact in simulation time for MPS, and for TTN this was probably a bit too naive. The code is not particularly easy to maintain and extend, so I am leaning towards deprecating it by "hiding it" and setting it to False by default.
- `use_kahypar` used to be a option in Config, but it makes more sense to set it here, since it is the only place where it is used. Moreover, it became clear that further work would be required to include KaHyPar in the default pipeline (due to dependencies), so it was already set to False by default.